### PR TITLE
Add documentation for CMS_EnvelopedData_create()

### PR DESCRIPTION
**Pull request by bukka**
Documentation for CMS_EnvelopedData_create() functions.

##### Checklist
- [x] documentation is added or updated

### CODE REVIEW

**Comment by slontis** [comment]
I am not sure why missingcrypto.txt doesn't complain.

**Comment by bukka** [comment]
Is there anything else that I should do in this PR? It's basically the base for the last doc thing in the CMS AES-GCM PR #8024 that I need to do so if someone could take a look, that would be awesome. :)

**Comment by beldmit** [comment]
LGTM.

File: doc/man3/CMS_EnvelopedData_create.pod [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+=head1 SYNOPSIS
+
+ #include <openssl/cms.h>
+
+ CMS_ContentInfo *CMS_EnvelopedData_create(const EVP_CIPHER *cipher);
+
+=head1 DESCRIPTION
+
+The B<CMS_ContentInfo>
```
Comment (mattcaswell) [inline doc/man3/CMS_EnvelopedData_create.pod:15]: This seems to be a half finished sentence

File: doc/man3/CMS_EnvelopedData_create.pod [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ #include <openssl/cms.h>
+
+ CMS_ContentInfo *CMS_EnvelopedData_create(const EVP_CIPHER *cipher);
+
+=head1 DESCRIPTION
+
+The B<CMS_ContentInfo>  
+
+CMS_EnvelopedData_create() creates a B<CMS_ContentInfo> structure with
+a type B<NID_pkcs7_enveloped>.
```
Comment (mattcaswell) [inline doc/man3/CMS_EnvelopedData_create.pod:18]: Please describe the `cipher` argument.

File: doc/man3/CMS_EnvelopedData_create.pod [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+The recipients can be added afterwards and the CMS_final() needs to be called
```
Comment (mattcaswell) [inline doc/man3/CMS_EnvelopedData_create.pod:20]: A reference to the function to add recipients would be a useful addition.

Comment (mattcaswell) [inline doc/man3/CMS_EnvelopedData_create.pod:20]: Remove "the" from in front of `CMS_final()`

File: doc/man3/CMS_EnvelopedData_create.pod [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+to finalize the B<CMS_ContentInfo> structure.
```
Comment (mattcaswell) [inline doc/man3/CMS_EnvelopedData_create.pod:21]: Probably we should also mention `CMS_ContentInfo_free`.

File: doc/man3/CMS_EnvelopedData_create.pod (hunk 1 above)
Comment (bukka) [reply doc/man3/CMS_EnvelopedData_create.pod]: Ah that should not be there. Not sure how it got there. Fixed.

File: doc/man3/CMS_EnvelopedData_create.pod (hunk 2 above)
Comment (bukka) [reply doc/man3/CMS_EnvelopedData_create.pod]: Done

File: doc/man3/CMS_EnvelopedData_create.pod (hunk 3 above)
Comment (bukka) [reply doc/man3/CMS_EnvelopedData_create.pod]: Done

File: doc/man3/CMS_EnvelopedData_create.pod (hunk 4 above)
Comment (bukka) [reply doc/man3/CMS_EnvelopedData_create.pod]: Done

File: doc/man3/CMS_EnvelopedData_create.pod (hunk 3 above)
Comment (bukka) [reply doc/man3/CMS_EnvelopedData_create.pod]: Added a reference in the same way as it is in `CMS_encrypt()` docs.

**Comment by bukka** [comment]
@mattcaswell all your comments hopefully addressed. Thanks for the review!

File: doc/man3/CMS_EnvelopedData_create.pod [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+=head1 DESCRIPTION
+
+CMS_EnvelopedData_create() creates a B<CMS_ContentInfo> structure with
+a type B<NID_pkcs7_enveloped>. B<cipher> is the symmetric cipher to use.
+
+The algorithm passed in the B<cipher> parameter must support ASN1 encoding of
+its parameters.
+
+The recipients can be added later using CMS_add1_recipient_cert() or
+CMS_add0_recipient_key().
```
Comment (mattcaswell) [inline doc/man3/CMS_EnvelopedData_create.pod:22]: For references to functions on other man pages enclose them in an `L` tag, with the section number (3) in parens, i.e. `L<CMS_add1_recipient_cert(3)>`

Comment (bukka) [reply doc/man3/CMS_EnvelopedData_create.pod:22]: Should this be done for all mentioned functions? I have been actually looking through the docs before and it seems that it's mostly without any link (see for example docs CMS_encrypt). I can of course add it but just wondering if I should add links everywhere...

Comment (mattcaswell) [reply doc/man3/CMS_EnvelopedData_create.pod:22]: Any function mentioned on  this page (except mentions of CMS_EnvelopedData_create itself) should be done in this way. There may well be discrepancies on other pages which don't do this - but this is the standard way we try to do it.

**Comment by mattcaswell** [comment]
Travis red cross appears unrelated.

**Comment by beldmit** [comment]
Merged. Thanks!

**Pull request closed without merging**